framework/console/controllers/AssetController.php

```diff
@@ -515,7 +515,7 @@ protected function compressJsFiles($inputFiles, $outputFile)
         if (is_string($this->jsCompressor)) {
             $tmpFile = $outputFile . '.tmp';
             $this->combineJsFiles($inputFiles, $tmpFile);
-            $this->stdout(shell_exec(strtr($this->jsCompressor, [
+            $this->stdout(shell_exec(strtr(Yii::getAlias($this->jsCompressor), [
                 '{from}' => escapeshellarg($tmpFile),
                 '{to}' => escapeshellarg($outputFile),
```

Comment:
Instead of `Yii::getAlias()` we can add `@alias` => `value` to the list of `strtr` arguments. That would solve the case @WinterSilence is talking about.

Reply:
@samdark can you explain your idea?

Reply:
I get @samdark's idea, but we would need to `strtr` the whole `BaseYii::$aliases` array.. not against it tho...

Reply:
@mikk150 why not extract this placeholders into property?

Reply:
@WinterSilence hmmmm... great questions :) I like those...

Reply:
well... because you cannot do this:

![image](https://user-images.githubusercontent.com/4953629/156770833-d29ad9ad-4fcf-491c-98fa-0d63f699bfb6.png)

Reply:
@mikk150 now just update your PR so I can do it